packages/now-next/src/index.ts

```diff
@@ -137,16 +143,35 @@ async function writeNpmRc(workPath: string, token: string) {
   );
 }
 
-function getNextVersion(packageJson: {
-  dependencies?: { [key: string]: string };
-  devDependencies?: { [key: string]: string };
-}) {
+/**
+ * Get the installed Next version.
+ *
+ * If you want to get the real installed Next version, you must first
+ * run npm install before calling this function
+ */
+async function getNextVersion(entryPath: string) {
   let nextVersion;
-  if (packageJson.dependencies && packageJson.dependencies.next) {
-    nextVersion = packageJson.dependencies.next;
-  } else if (packageJson.devDependencies && packageJson.devDependencies.next) {
-    nextVersion = packageJson.devDependencies.next;
+
+  try {
+    // First try to resolve the `next` dependency and get the real version from its
+    // package.json. This allows the builder to be used with frameworks like Blitz that
+    // bundle Next but where Next isn't in the project root's package.json
+    nextVersion = require(resolveFrom(entryPath, 'next/package.json')).version;
+
+    debug(`Detected Next.js version: ${nextVersion}`);
+  } catch (_ignored) {
+    debug(
+      `Could not identify real Next.js version, falling back to check root package.json`
+    );
+
+    const pkg = await readPackageJson(entryPath);
+    if (pkg.dependencies && pkg.dependencies.next) {
+      nextVersion = pkg.dependencies.next;
+    } else if (pkg.devDependencies && pkg.devDependencies.next) {
+      nextVersion = pkg.devDependencies.next;
+    }
```

Comment:
This should clean or coerce the semver version since it's now being used below where we expect an absolute version and not one with semver content.

Reply:
@Timer I checked all the places `nextVersion` is being used, and every place is using the `semver` package to read it.

Are you sure this needs changed here? If so, looks like I would simple wrap with `semver.clean()`?